### DISCUSSION TITLE
Detect if program is currently running on a Turtle or a Computer?

### DISCUSSION THREAD

**Discussion by CarelessBF** [cc-tweaked/CC-Tweaked#1858] [Q&A] [answered]
Is there a way to automatically detect if the current program is running on a Turtle or computer?
I tried to look around myself and couldn't find a solution.

I'm attempting to set up a OOP Class system via a datapack but I need a way to distinguish turtles and computers without inputting it manually and without using peripheral.getType(), if possible obviously.

Context:
Using the Datapack method, I'm aiming to create a OOP Class system so that computers and turtles can inherit functions and attributes , reducing the amount of redundant files/code.
The datapack method I'm referring to is this: https://github.com/cc-tweaked/datapack-example
Currently, my classes.lua file is in "data\computercraft\lua\rom\modules\main" file path but I'm not sure if this is the correct place for it.

I suspect a possible solution having to do with the fact that only turtles/computers can see files in certain directories in the datapack files but I can piece a solution together that wouldn't result in errors.

It's been difficult to find source code of projects with a similar idea of mine (with OOP Classes and such), but I'm sure its been done before. 
I'm new to all of this and I would love to hear your suggestions :)

**Comment by SquidDev** [comment, marked as answer]
The `turtle` API is only present on turtles, so you can just check for its presence. 

```lua 
if turtle then
  print("On a turtle")
else
  print("On something else")
end

**Reply by CarelessBF** [reply]
What seems to be the most difficult problem always has the most simple solutions lol. How ironic! I appreciate the quick response!
